src/Native/FactorizationMachineNative/FactorizationMachineCore.cpp

```diff
@@ -0,0 +1,175 @@
+#include <cmath>
+#include <cstring>
+#include <limits>
+#include <pmmintrin.h>
+#include "../Stdafx.h"
+
+// Compute the output value of the field-aware factorization, as the sum of the linear part and the latent part. 
+// The linear part is the inner product of linearWeights and featureValues.
+// The latent part is the sum of all intra-field interactions in one field f, for all fields possible. 
+EXPORT_API(void) CalculateIntermediateVariablesNative(int fieldCount, int latentDim, int count, _In_ int * fieldIndices, _In_ int * featureIndices, _In_ float * featureValues,
+    _In_ float * linearWeights, _In_ float * latentWeights, _Inout_ float * latentSum, _Out_ float * response)
+{
+    // The number of all possible fields
```

Comment:
poor choice for variable names. Anything more mnemonic?

Reply:
We discussed about this before. Those symbols are close to my pseudo code in a pdf so that anyone who read the document first can get familiar with them. On the other hand, even with meaningful names, it's almost impossible to understand SSE code for users.